pkg/auth/providers/common/provider_util_test.go

```diff
@@ -41,3 +46,66 @@ func TestEscapeUUID(t *testing.T) {
 		})
 	}
 }
+
+func TestDecode(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name    string
+		input   any
+		output  any
+		wantErr bool
+	}{
+		{
+			name:    "successful decode",
+			input:   getMockAuthConfig(),
+			output:  &apimgmtv3.AuthConfig{},
+			wantErr: false,
+		},
+		{
+			name:    "unsuccessful decoder create",
+			input:   getMockAuthConfig(),
+			output:  apimgmtv3.AuthConfig{},
+			wantErr: true,
+		},
+		{
+			name:    "unsuccessful decode",
+			input:   "bogus input",
+			output:  &apimgmtv3.AuthConfig{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
Nit (feel free to ignore): you could run the subtests also in parallel, assuming you don't have any shared state among them.

```
for _, tt := range tests {
        tt := tt   // to avoid all goroutines running the last element of the collection (Go's loop and goroutines quirk)
	t.Run(tt.name, func(t *testing.T) {
                t.Parallel()
```

Reply:
probably minimal -> zero impact at this point in time, but yeah, could be done.